crates/ra_assists/src/handlers/create_function.rs

```diff
@@ -0,0 +1,313 @@
+use ra_syntax::{
+    ast::{self, AstNode},
+    SmolStr, SyntaxKind, SyntaxNode, TextUnit,
+};
+
+use crate::{Assist, AssistCtx, AssistId};
+use ast::{CallExpr, Expr};
+use ra_fmt::leading_indent;
+
+// Assist: create_function
+//
+// Creates a stub function with a signature matching the function under the cursor.
+//
+// ```
+// fn foo() {
+//      bar<|>("", baz());
+// }
+//
+// ```
+// ->
+// ```
+// fn foo() {
+//      bar("", baz());
+// }
+//
+// fn bar(arg_1: &str, baz: Baz) {
+//      todo!();
+// }
+//
+// ```
+pub(crate) fn create_function(ctx: AssistCtx) -> Option<Assist> {
+    let path: ast::Path = ctx.find_node_at_offset()?;
+
+    if ctx.sema.resolve_path(&path).is_some() {
+        // The function call already resolves, no need to create a function
+        return None;
+    }
+
+    if path.qualifier().is_some() {
+        return None;
+    }
+
+    let call: ast::CallExpr = ctx.find_node_at_offset()?;
+
+    let (generated_fn, cursor_pos, text_start) = generate_fn(&ctx, &call)?;
+
+    ctx.add_assist(AssistId("create_function"), "Create function", |edit| {
```

Comment:
NIT: maybe "add" or "insert" instead of "create" to be consistent with other assists.

Reply:
Yeah, let's use `add`, but I guess one day we should really rename all similar things to `generate`.

Reply:
Will do

Reply:
Done